File: dotnet/DesignPatterns/StructuralDesignPatterns/ProxyDesignPattern/ProxyDesignPattern.RemoteProxyExample/IPaymentService.cs
namespace ProxyDesignPattern.RemoteProxyExample;

/// <summary>
/// Subject Interface - Common interface for payment operations.
/// </summary>
public interface IPaymentService
{
    PaymentResult ProcessPayment(decimal amount, string cardNumber);
    decimal GetBalance(string accountId);
}

/// <summary>
/// Represents the result of a payment operation.
/// </summary>
public record PaymentResult(bool Success, string TransactionId, string Message);

File: nodejs/DesignPatterns/StructuralDesignPatterns/ProxyDesignPattern/ProxyDesignPattern.RemoteProxyExample/src/IPaymentService.ts
/**
 * Subject Interface - Common interface for payment operations.
 */
export interface IPaymentService {
  processPayment(amount: number, cardNumber: string): Promise<PaymentResult>;
  getBalance(accountId: string): Promise<number>;
}

/**
 * Represents the result of a payment operation.
 */
export interface PaymentResult {
  success: boolean;
  transactionId: string;
  message: string;
}
